### ISSUE TITLE
create MyModel class

### ISSUE THREAD

**Issue by PeterMitrano** [UM-ARM-Lab/mps_shape_completion#13]
- inherit from tf.keras.Model
 - call method. needs to return everything nessecary for computing loss and some metrics
 - train_step method
 - optimizers member variable, defaults to Adam
 - hparams variable
 - apply_gradients method, returns dictionary of metrics
 - calc_metrics function, takes in dataset element and output of call, returns a dictionary

**Comment by PeterMitrano** [comment]
#24
